src/runtime-helpers.js

```diff
@@ -802,9 +801,61 @@ function setPaths (flags) {
   return filecomponents
 }
 
+/**
+ * Handle Adobe auth action dependency
+ *
+ * This is a temporary solution and needs to be removed when headless apps will be able to
+ * validate against app-registry
+ *
+ * This function stores the IMS organization id in the Adobe I/O cloud state library which
+ * is required by the headless validator.
+ *
+ * The IMS org id must be stored beforehand in `@adobe/aio-lib-core-config` under the
+ * `'project.org.ims_org_id'` key.
+ */
+async function setupAdobeAuth (actions, owOptions) {
+  // do not modify those
+  const ADOBE_HEADLESS_AUTH_ACTION = '/adobeio/shared-validators-v1/headless'
+  const AIO_STATE_KEY = '__aio'
+  const AIO_CONFIG_IMS_ORG_ID = 'project.org.ims_org_id'
+  const AIO_STATE_PUT_ENDPOINT = 'https://adobeio.adobeioruntime.net/api/v1/web/state/put'
+
+  const hasAnAdobeHeadlessAuthSequence = actions.some(a => a.exec && a.exec.kind === 'sequence' && a.exec.components.includes(ADOBE_HEADLESS_AUTH_ACTION))
+  if (hasAnAdobeHeadlessAuthSequence) {
+    // if we use the headless (default auth action) we need to store the ims org id in the
+    // cloud state lib. This is needed by the auth action to perform an org check.
+    const imsOrgId = require('@adobe/aio-lib-core-config').get(AIO_CONFIG_IMS_ORG_ID)
```

Comment:
any advantages here to require inside this scope versus global scope for clarity?

Reply:
kept it like this cause this code is not supposed to stay there for long but yeah will move to the global scope

Reply:
ok moved the aio config to global scope but left node fetch require into local scope as it was breaking some tests + node fetch shouldn't be useful in the rt plugin outside of this workaround